Service/src/main/java/ticket/Ticket.java

```diff
@@ -0,0 +1,253 @@
+        this.stadiumSector = '\u0000';
+        this.maxWeight = 0;
+        this.cost = cost;
+        this.ticketType = ticketType;
+    }
+
+    public Ticket(short eventCode, String concertHall, LocalDate day, LocalTime time,
+                  boolean isPromo, char stadiumSector, double maxWeight, BigDecimal cost, int ticketType){
+        this.ID = generateId();
+        if(eventCode<100 || eventCode>999) {
+            throw new IllegalArgumentException("Enter 3-digit code");
+        }
+        this.eventCode = eventCode;
+        this.creationDate = LocalDate.now();
+        this.creationTime = LocalTime.now();
+        if(concertHall.length()>10){
+            throw new IllegalArgumentException("The name of the concert hall should not exceed 10 characters");
+        }
+        this.concertHall = concertHall;
+        this.day = day;
+        this.time = time;
+        if(isPromo){
+            this.isPromo = ticket.PromotionAvaliabilities.YES;
+
+        }else{
+            this.isPromo = ticket.PromotionAvaliabilities.NO;
+
+        }
+        if(stadiumSector != 'A' && stadiumSector != 'B' && stadiumSector != 'C'){
+            throw new IllegalArgumentException("The name of sectors must be A, B or C");
+        }
+        this.stadiumSector = stadiumSector;
+        this.maxWeight = maxWeight;
+        this.cost = cost;
+        if(ticketType == 0){
+            this.ticketType = ticketTypes.NOINFO;
+        }else if(ticketType == 1){
+            this.ticketType = ticketTypes.DAY;
+        }else if(ticketType == 2){
+            this.ticketType = ticketTypes.WEEK;
+        }else if(ticketType == 3){
+            this.ticketType = ticketTypes.MONTH;
+        }else if(ticketType == 4){
+            this.ticketType = ticketTypes.YEAR;
+
+        }
+    }
+
+    public char[] getTicketId(){
+        return this.ID;
+    }
+
+    public short getTicketEvetCode(){
+        return this.eventCode;
+    }
+
+    public LocalDate getTicketCreationDate(){
+        return this.creationDate;
+    }
+
+    public LocalTime getTicketCreationTime(){
+        return this.creationTime;
+    }
+
+    public String getTicketConcertHall(){
+        return this.concertHall;
+    }
+
+    public LocalDate getTicketDay(){
+        return this.day;
+    }
+
+    public LocalTime getTicketTime(){
+        return this.time;
+    }
+
+    public PromotionAvaliabilities getTicketIsPromo(){
+        return this.isPromo;
+    }
+
+    public char getTicketStadiumSector(){
+        return this.stadiumSector;
+    }
+
+    public double getTicketMaxWeight(){
+        return this.maxWeight;
+    }
+
+    public BigDecimal getTicketCost(){
+        return this.cost;
+    }
+
+    public void setStadiumSector(char sector){
+        this.stadiumSector = sector;
+    }
+
+    public void setTicketDay(LocalDate date){
+        this.day = date;
+    }
+
+    public void setTicketTime(LocalTime time){
+        this.time = time;
+    }
+
+    private char[] iDGenerator(){
```

Comment:
IdGenerator is not responsibility of the model class, move it please to separate class